src/test/java/com/notably/model/block/BlockTreeTest.java

```diff
@@ -0,0 +1,104 @@
+package com.notably.model.block;
+
+import static com.notably.testutil.Assert.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.notably.commons.core.path.AbsolutePath;
+import com.notably.commons.core.path.exceptions.InvalidPathException;
+import com.notably.model.block.exceptions.CannotModifyRootException;
+import com.notably.model.block.exceptions.DuplicateBlockException;
+import com.notably.model.block.exceptions.NoSuchBlockException;
+
+public class BlockTreeTest {
```

Comment:
In addition, I don't think you've tested any observability behavior here. Perhaps you'd like to add test cases on that?